lib/engines/content_object_store/app/models/content_object_store/content_block/edition/updatable.rb

```diff
@@ -0,0 +1,38 @@
+module ContentObjectStore
+  module ContentBlock::Edition::Updatable
+    extend ActiveSupport::Concern
+
+    def create_new_edition(edition_params:)
+      new_edition = dup.tap do |edition|
+        edition.state = :draft
+        edition.creator = creator
+        edition.document = document
+        edition.assign_attributes(
+          filter_params_for_validation_check(edition_params),
+        )
+      end
+
+      unless new_edition.valid?
+        raise ActiveRecord::RecordInvalid, new_edition
+      end
+
+      new_edition.save!
```

Comment:
I think these are all points to consider before moving forwards:

1. I think we need to remember to assign the new edition as the documents `latest_edition_id`. Having that close to the point of creation feels important so that it isn't forgotten.

    We have this happening over in this [CreateEdition service](https://github.com/alphagov/whitehall/blob/9feec79e7c0205cd386c4162d7aaed5627f086a3/lib/engines/content_object_store/app/services/content_object_store/create_edition_service.rb#L11). This class also feels relevant to this change as it has a different way of doing the same(?) thing so;

2. It seems to me like `CreateEditionService` and `UpdateEditionService` would both benefit from using the same method?

3. When we consider the above point 2: the naming of 'updatable' feels off since creating isn't updating. How about 'creatable' as a slight tweak? It goes play nicely with the method name we have of "create_new_edition"

I feel like I'm losing my grasp of what is a Publishable and what would be a Creatable concern. Do you have a definition for how you think about it? Could it be Publishable is about downstream requests (to publishing API or to our queue) and Creatable is only about inside Whitehall? I think I'd value some sort of definition for each concern so I know how to approach each file. What do you think?

Reply:
To me, `Creatable` doesn't feel the right name either because we're making a copy of the Edition and then returning a new one, but then, neither does `Updatable`. Maybe we need to rethink the API altogether. Maybe it would be better to have a class level method that takes the latest edition and the updated params?

This might help with `2` because using it in the context of creating the first edition would be easier because we could make the latest edition optional?

Reply:
If it's called on an edition the method could just be create_copy

Reply:
How about `Copyable` for the name of the concern? I like the idea of having a `update_document_reference_to_latest_edition!` method. I know callbacks aren't usually the best idea, because they hide away, but this does feel like a good use case for them. I'll have a think...

Reply:
I've had a think, and no. I remember why I hate callbacks 😆

Reply:
Callbacks does feel like the Rails way. I too pushed them away due to their behaviour being hidden and magical.